File: PROYECTO 1/airplane.cpp
#include "airplane.h"

using namespace std;

// Constructor
Airplane::Airplane(string brand, string model, int capacity, int maxSpeed,
    int autonomy, int year, string condition, string ubication,
    bool availability, int maxAltitude, int engineAmount, string category)
    : Aircraft(brand, model, capacity, maxSpeed, autonomy, year, condition,
    ubication, availability), maxAltitude(maxAltitude),
    engineAmount(engineAmount), category(category) {}

// Getters y setters para los atributos específicos de Airplane
int Airplane::getMaxAltitude()  { return maxAltitude; }
int Airplane::getEngineAmount()  { return engineAmount; }
string Airplane::getCategory()  { return category; }

void Airplane::setMaxAltitude(int newMaxAltitude) { maxAltitude = newMaxAltitude; }
void Airplane::setEngineAmount(int newEngineAmount) { engineAmount = newEngineAmount; }
void Airplane::setCategory(string newCategory) { category = newCategory; }

File: pythonPr/airplane.py
from aircraft import Aircraft

class Airplane(Aircraft):
    def __init__(self,  brand = None,  model = None,  ident = None,  capacity = None,  max_speed = None,  autonomy = None,  year = None,  condition = None, max_altitude = None,  engine_amount = None,  category = None):
        super().__init__(brand, model, ident, capacity, max_speed, autonomy, year, condition)
        self.max_altitude = max_altitude
        self.engine_amount = engine_amount
        self.category = category

    def get_max_altitude(self):
        return self.max_altitude

    def get_engine_amount(self):
        return self.engine_amount

    def get_category(self):
        return self.category

    def set_max_altitude(self, new_max_altitude):
        self.max_altitude = new_max_altitude

    def set_engine_amount(self, new_engine_amount):
        self.engine_amount = new_engine_amount

    def set_category(self, new_category):
        self.category = new_category